### PR TITLE
mrc-4232 test integration between front-end and API

### PR DESCRIPTION
**Pull request by hillalex**
Adds a jest test suite for testing that the back-end is serving responses as expected, and that the front-end is able to receive and parse them. This also implicitly tests that the scheduler is running (because packet data is returned as expected).

### CODE REVIEW

File: app/src/tests/integration/actions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        const asyncThunk = api.get<Packet[]>(PacketsMutationType.GetPackets, "/bad-url")();
+        await asyncThunk(dispatch, jest.fn(), jest.fn());
+
+        expect(dispatch.mock.calls[0][0]).toMatchObject({
+            type: "GetPackets/pending",
+            payload: undefined,
+        });
+        const result = dispatch.mock.calls[1][0];
+        expect(result["type"]).toBe("GetPackets/rejected");
+        expect(result["payload"]["error"]).toBe("Not Found");
```
Comment (hillalex) [inline app/src/tests/integration/actions.test.tsx:19]: this comes from the built in Spring error format, open ticket to do custom error handling in the API

**Comment by hillalex** [comment]
I cannot really explain what codecov is on about.

File: app/package.json [hunk 2]
```diff
@@ -39,12 +39,14 @@
   "jest": {
     "moduleNameMapper": {
       "^axios$": "axios/dist/axios.js"
-    }
+    },
+    "coveragePathIgnorePatterns": ["/node_modules/", "tests"]
```
Comment (EmmaLRussell) [inline app/package.json:43]: Should this be `src/tests`? or is `coveragePathIgnorePatterns` sensitive to /s?

File: .github/workflows/frontend-test.yml [hunk 3]
```diff
@@ -11,6 +11,9 @@ on:
   pull_request:
     branches:
       - 'main'
+env:
+  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
```
Comment (EmmaLRussell) [inline .github/workflows/frontend-test.yml:15]: What's this getting used for?

**Review by EmmaLRussell — COMMENTED** [review]
LGTM

File: app/package.json (hunk 2 above)
Comment (hillalex) [reply app/package.json:43]: it's a regex so this will match "tests" anywhere in the path

File: .github/workflows/frontend-test.yml (hunk 3 above)
Comment (hillalex) [reply .github/workflows/frontend-test.yml:15]: `api/scripts/run` runs the dockerised api using the current branch name as a tag

**Pull request merged**